cpp/usb_worker.cpp

```diff
@@ -91,11 +91,6 @@ static std::string maybe_get_string(libusb::Device & dev, uint8_t string, int re
         auto err = std::get<libusb::Error>(maybe_devh);
         switch (err.number) {
         case LIBUSB_ERROR_ACCESS:
-            if (retry <= 0) {
-                return err.get_message();
-            }
-            std::this_thread::sleep_for(std::chrono::milliseconds(500));
-            return maybe_get_string(dev, string, retry - 1);
```

Comment:
The problem isn't that we get a serial = "LIBUSB_ERROR_ACCESS", since sometimes that is indeed the correct "state" of the interface. We poll again and again regardless, from what I gather the problem is that we don't update the serial.
In any case, when calling maybe_get_string we filter results where serial = "LIBUSB_ERROR_ACCESS":
https://github.com/Huddly/device-api-usb/blob/03cb658d5cbc415789e16c686891581ecdec4cd7/cpp/usb_worker.cpp#L186

Reply:
I understand, but I would argue that is never correct that ```maybe_get_string``` returns that as the serial string.